### PR TITLE
Fix flakyness of replace node e2e

### PR DESCRIPTION
**Pull request by zimnx**
Full quorum check is sensitive to seeing nodes not being part of the cluster, when such node is spotted it immediately returns an error.
In replace node e2e, it could happen that cluster become ready before replaced node was acknowledged by the others.
In this case, full quorum check returned an error, because it saw a IP not bound to any cluster Service.

To fix it, test waits until cluster is rolled out, and replaced node is seen by the other one, before the full quorum check is validated.

### CODE REVIEW

**Review by tnozicka — COMMENTED** [review]
would the readiness check @rzetelskik is building help here?

**Comment by zimnx** [comment]
> would the readiness check @rzetelskik is building help here?

AFAIR it allows for one DN, we have 2 nodes in this e2e, so it wouldn't help.

**Comment by rzetelskik** [comment]
> AFAIR it allows for one DN, we have 2 nodes in this e2e, so it wouldn't help.

Yes, it wouldn't help in this case. After it's been merged you could just increase the number of nodes in this test to 3 though.

File: test/e2e/set/scyllacluster/scyllacluster_replace.go [hunk 1]
```diff
@@ -86,6 +86,25 @@ var _ = g.Describe("ScyllaCluster replace", func() {
...
+		otherNodeService, err := f.KubeClient().CoreV1().Services(sc.Namespace).Get(ctx, utils.GetNodeName(sc, 1), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		o.Eventually(func(g o.Gomega) {
+			status, err := client.Status(ctx, otherNodeService.Spec.ClusterIP)
+			g.Expect(err).NotTo(o.HaveOccurred())
+			g.Expect(status.LiveHosts()).To(o.ContainElement(replacedNodeService.Spec.ClusterIP))
+		}).WithPolling(time.Second).WithTimeout(5 * time.Minute).Should(o.Succeed())
+
+		framework.By("Replace has been acknowledged by the other node")
```
Comment (tnozicka) [inline test/e2e/set/scyllacluster/scyllacluster_replace.go:106]: this doesn't read well. it should be either a separate framework.info but usually the "next" message will imply waiting was finished

Comment (zimnx) [reply test/e2e/set/scyllacluster/scyllacluster_replace.go]: removed

**Review by tnozicka — APPROVED** [review]
/lgtm
/approve

**Review by tnozicka — APPROVED** [review]

**Pull request merged**